python/graphstorm/config/argument.py

```diff
@@ -1707,6 +1723,28 @@ def eval_metric(self):
 
         return eval_metric
 
+    @property
+    def model_select_etype(self):
+        """ Canonical etype used for selecting the best model
+        """
+        # pylint: disable=no-member
+        if hasattr(self, "_model_select_etype"):
+            etype = self._model_select_etype.split(",")
+            if len(etype) == 1:
+                assert etype[0].upper() == LINK_PREDICTION_MAJOR_EVAL_ETYPE_ALL, \
+                    "If you want to select model based on the average eval value of" \
+                    "all edge types please set model_select_etype to " \
+                    f"{LINK_PREDICTION_MAJOR_EVAL_ETYPE_ALL}"
+                return LINK_PREDICTION_MAJOR_EVAL_ETYPE_ALL
```

Comment:
is this the same as not specifying it? if so, why do we add additional rule?

Reply:
Yes.
Let me remove it.